### PR TITLE
Add Trino 406 release notes

### LINKED ISSUE #15563: Release notes for 406

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 406 release. Fixes #15563  

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

## Verification for each pull request

Format: maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 29 Dec 2022

@hashhar #15542 ✅ rn ✅ docs
@ebyhr #15553 ✅ rn ✅ docs

## 30 Dec 2022

@ebyhr #15538 ✅ rn ✅ docs
@ebyhr #15566 ✅ rn ✅ docs
@ebyhr #15570 ✅ rn ✅ docs

## 2 Jan 2023

@ebyhr #15544 ✅ rn ✅ docs

## 3 Jan 2023

@findepi #15573 ✅ rn ✅ docs
@findepi #15574 ✅ rn ✅ docs
@raunaqmorarka #15552 ✅ rn ✅ docs
@sopel39 #15578 ✅ rn ✅ docs
@ebyhr #15565 ✅ rn ✅ docs

## 4 Jan 2023

@losipiuk #15509 ✅ rn ✅ docs
@Praveen2112 #15550 ✅ rn ✅ docs
@sopel39 #15582 ✅ rn ✅ docs
@raunaqmorarka #15554 ✅ rn ✅ docs
@raunaqmorarka #15498 ✅ rn ✅ docs
@findepi #15581 ✅ rn ✅ docs
@Praveen2112 #15589 ✅ rn ✅ docs

## 5 Jan 2023

@losipiuk #15160 ✅ rn ✅ docs (#15644)
@findepi #15602 ✅ rn ✅ docs
@losipiuk #15497 ✅ rn ✅ docs
@findepi #15611 ✅ rn ✅ docs
@findepi #15610 ✅ rn ✅ docs
@arhimondr #15601 ✅ rn ✅ docs
@kokosing #15595 ✅ rn ✅ docs
@ebyhr #15621 ✅ rn ✅ docs
@ebyhr #15607 ✅ rn ✅ docs
@raunaqmorarka #15608 ✅ rn ✅ docs

## 6 Jan 2023

@raunaqmorarka #15269 ✅ rn ✅ docs
@raunaqmorarka #14428 ✅ rn ✅ docs

## 9 Jan 2023

@arhimondr #15624 ✅ rn ✅ docs
@arhimondr #15632 ✅ rn ✅ docs
@ebyhr #15434 ✅ rn ✅ docs
@ebyhr #15591 ✅ rn ✅ docs
@hashhar #15647 ✅ rn ✅ docs

## 10 Jan 2023

@kasiafi #15640 ✅ rn ✅ docs
@findepi #15635 ✅ rn ✅ docs
@ebyhr #15517 ✅ rn ✅ docs
@raunaqmorarka #15642 ✅ rn ✅ docs
@raunaqmorarka #15428 ✅ rn ✅ docs
@arhimondr #15402 ✅ rn ✅ docs
@arhimondr #15514 ✅ rn ✅ docs

## 11 Jan 2023

@Praveen2112 #15594 ✅ rn ✅ docs
@findepi #15609 ✅ rn ✅ docs
@findepi #15619 ✅ rn ✅ docs
@findepi #15618 ✅ rn ✅ docs
@arhimondr #15627 ✅ rn ✅ docs
@arhimondr #15662 ✅ rn ✅ docs
@hashhar #15671 ✅ rn ✅ docs
@dain #15675 ✅ rn ✅ docs
@ebyhr #15655 ✅ rn ✅ docs

## 12 Jan 2023

@hashhar #15658 ✅ rn ✅ docs
@losipiuk #14597 ✅ rn ✅ docs
@findepi #15657 ✅ rn ✅ docs
@findepi #15670 ✅ rn ✅ docs
@arhimondr #15693 ✅ rn ✅ docs
@ebyhr #15699 ✅ rn ✅ docs

## 13 Jan 2023

@ebyhr #15674 ✅ rn ✅ docs
@sopel39 #15695 ✅ rn ✅ docs
@Praveen2112 #15691 ✅ rn ✅ docs
@findepi #15694 ✅ rn ✅ docs
@arhimondr #15644 ✅ rn ✅ docs
@Praveen2112 #15692 ✅ rn ✅ docs
@martint #15698 ✅ rn ✅ docs
@martint #15169 ✅ rn ✅ docs
@dain #15718 ✅ rn ✅ docs
@raunaqmorarka #15706 ✅ rn ✅ docs

## 14 Jan 2023

@Praveen2112 #15688 ✅ rn ✅ docs

## 15 Jan 2023

@ebyhr #15681 ✅ rn ✅ docs
@ebyhr #15719 ✅ rn ✅ docs

## 16 Jan 2023

@hashhar #15725 ✅ rn ✅ docs
@kasiafi #15653 ✅ rn ✅ docs
@sopel39 #15711 ✅ rn ✅ docs
@ebyhr #15717 ✅ rn ✅ docs
@findepi #15710 ✅ rn ✅ docs
@sopel39 #15735 ✅ rn ✅ docs
@ebyhr #15707 ✅ rn ✅ docs
@arhimondr #15678 ✅ rn ✅ docs
@ebyhr #15727 ✅ rn ✅ docs
@raunaqmorarka #15728 ✅ rn ✅ docs
@ebyhr #15551 ✅ rn ✅ docs

## 17 Jan 2023

@hashhar #15739 ✅ rn ✅ docs
@ebyhr #15648 ✅ rn ✅ docs
@findepi #15576 ✅ rn ✅ docs
@findepi #15730 ✅ rn ✅ docs
@findepi #15548 ✅ rn ✅ docs
@arhimondr #15616 ✅ rn ✅ docs
@electrum #15720 ✅ rn ✅ docs
@martint #15680 ✅ rn ✅ docs
@martint #15465 ✅ rn ✅ docs
@phd3 #15017 ✅ rn ✅ docs
@electrum #15704 ✅ rn ✅ docs

## 18 Jan 2023

@findepi #15709 ✅ rn ✅ docs
@hashhar #15684 ✅ rn ✅ docs
@findepi #15673 ✅ rn ✅ docs
@findepi #15615 ✅ rn ✅ docs
@arhimondr #15672 ✅ rn ✅ docs
@electrum #15754 ✅ rn ✅ docs
@electrum #15630 ✅ rn ✅ docs
@ebyhr #15651 ✅ rn ✅ docs
@electrum #15161 ✅ rn ✅ docs
@raunaqmorarka #15742 ✅ rn ✅ docs
@ebyhr #15758 ✅ rn ✅ docs
@ebyhr #11772 ✅ rn ✅ docs

## 19 Jan 2023

@findepi #15756 ✅ rn ✅ docs
@findepi #15759 ✅ rn ✅ docs
@raunaqmorarka #15569 ✅ rn ✅ docs
@arhimondr #15766 ✅ rn ✅ docs

## 20 Jan 2023

@losipiuk #15761 ✅ rn ✅ docs
@findepi #15783 ✅ rn ✅ docs
@findepi #15757 ✅ rn ✅ docs
@sopel39 #15799 ✅ rn ✅ docs
@arhimondr #15764 ✅ rn ✅ docs
@losipiuk #15620 ✅ rn ✅ docs
@arhimondr #15715 ✅ rn ✅ docs
@ebyhr #15796 ✅ rn ✅ docs
@ebyhr #15786 ✅ rn ✅ docs

## 22 Jan 2023

@sopel39 #15767 ✅ rn ✅ docs
@losipiuk #15804 ✅ rn ✅ docs
@ebyhr #15805 ✅ rn ✅ docs

## 23 Jan 2023

@findepi #15782 ✅ rn ✅ docs
@sopel39 #15686 ✅ rn ✅ docs
@electrum #15494 ✅ rn ✅ docs
@electrum #15530 ✅ rn ✅ docs
@electrum #15597 ✅ rn ✅ docs
@electrum #15599 ✅ rn ✅ docs
@electrum #15600 ✅ rn ✅ docs
@electrum #15404 ✅ rn ✅ docs
@electrum #14768 ✅ rn ✅ docs
@electrum #15623 ✅ rn ✅ docs
@electrum #15033 ✅ rn ✅ docs
@electrum #15225 ✅ rn ✅ docs
@martint #15798 ✅ rn ✅ docs
@sopel39 #15815 ✅ rn ✅ docs
@arhimondr #15803 ✅ rn ✅ docs
@phd3 #13243 ✅ rn ✅ docs
@ebyhr #15776 ✅ rn ✅ docs

## 24 Jan 2023

@findepi #15814 ✅ rn ✅ docs
@raunaqmorarka #15760 ✅ rn ✅ docs
@sopel39 #15708 ✅ rn ✅ docs
@findepi #15824 ✅ rn ✅ docs
@dain #15818 ✅ rn ✅ docs
@ebyhr #15800 ✅ rn ✅ docs
@ebyhr #14964 ✅ rn ✅ docs
@ebyhr #15831 ✅ rn ✅ docs

## 24 Jan 2023

@sopel39 #15834 ✅ rn ✅ docs
@sopel39 #15835 ✅ rn ✅ docs
@ebyhr #15808 ✅ rn ✅ docs
@arhimondr #15696 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release.rst [hunk 1]
```diff
@@ -10,6 +10,7 @@ Release notes
 .. toctree::
     :maxdepth: 1
```
Comment (mosabua) [inline docs/src/main/sphinx/release.rst:12]: .. _releases_2022:

```
2023
====

.. toctree::
    :maxdepth: 1

```

is missing

File: docs/src/main/sphinx/release/release-406.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  enabled. ({issue}`15160`)
+
+## BigQuery connector
+
+* Fix possible incorrect results for certain queries like `COUNT(*)` when a
+  table has recently been written to. ({issue}`14981`)
+
+## Delta Lake connector
+
+* Improve performance of reading from Parquet files. ({issue}`15498`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:15]:
```suggestion
* Improve performance of reading Parquet files. ({issue}`15498`)
```

File: docs/src/main/sphinx/release/release-406.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+* Improve memory accounting for the Parquet reader. ({issue}`15554`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:16]:
```suggestion
* Improve memory accounting of the Parquet reader. ({issue}`15554`)
```

File: docs/src/main/sphinx/release/release-406.md (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-406.md:15]: same for other entries

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release.rst [hunk 4]
```diff
@@ -2,6 +2,16 @@
 Release notes
 *************
 
+.. _releases_2023:
+
+2023
```
Comment (mosabua) [inline docs/src/main/sphinx/release.rst:7]: ;-)

File: docs/src/main/sphinx/release/release-406.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
+# Release 406 (6 Jan 2023)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:1]: 11

File: docs/src/main/sphinx/release/release-406.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+## General
+
+* Add support for exchange spooling on HDFS when fault-tolerant execution is
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:5]: Lets make sure the doc PR for this is merged and maybe even link to that section here. I kinda think that is what we did when we added the others

File: docs/src/main/sphinx/release/release-406.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  enabled. ({issue}`15160`)
+* Improve planner estimates for queries containing outer joins over a `topN`
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:7]: Should it really be code highlighted and formatted like that? Or should we use what is displayed in a query plan without code highlight?

File: docs/src/main/sphinx/release/release-406.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+  operation. ({issue}`15428`)
+* Fix inaccurate planner estimates for queries with filters on columns without
+  statistics. ({issue}`15642`)
+* Fix queries with outer joins failing when fault-tolerant execution is
+  enabled. ({issue}`15608`)
+* Fix potential query failure when using `MATCH_RECOGNIZE`. ({issue}`15461`)
+
+## BigQuery connector
+
+* Fix possible incorrect results for certain queries like `COUNT(*)` when a
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:17]: count 

functions are lowercase..

File: docs/src/main/sphinx/release/release-406.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+## Iceberg connector
+
+* Improve performance of reading Parquet files. ({issue}`15498`)
+* Improve memory accounting of the Parquet reader. ({issue}`15554`)
+* Improve performance of queries with filters or projections on low-cardinality
+  string columns stored in Parquet files. ({issue}`15269`)
+* Fix bug which could lead to more data than necessary getting read from Parquet
+  files for queries with filters. ({issue}`15552`)
+* Fix query failure when a subquery contains [time travel](iceberg-time-travel). ({issue}`15607`)
+
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:63]: Remove empty line

File: docs/src/main/sphinx/release/release-406.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+## PostgreSQL connector
+
+* Add support for changing column types. ({issue}`15515`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:67]: Empty line at eof

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-406.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+  operation. ({issue}`15428`)
+* Remove `legacy-phased` execution scheduler. ({issue}`15657`)
+* Fix failure when `WHERE` or `JOIN` clauses contain a `LIKE` expression with a
+  non-constant pattern or escape. ({issue}`15629`)
+* Fix inaccurate planner estimates for queries with filters on columns without
+  statistics. ({issue}`15642`)
+* Fix queries with outer joins failing when fault-tolerant execution is
+  enabled. ({issue}`15608`)
+* Fix potential query failure when using `MATCH_RECOGNIZE`. ({issue}`15461`)
+* Fix query failure when column masking or a row filter involves columns. ({issue}`15583`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-406.md:17]: The issue seem different from what this entry describes - it appears to be related to groups

File: docs/src/main/sphinx/release/release-406.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+* Fix possible incorrect results for certain queries like `count(*)` when a
+  table has recently been written to. ({issue}`14981`)
+
+## Delta Lake connector
+
+* Improve performance of reading Parquet files. ({issue}`15498`)
+* Improve memory accounting of the Parquet reader. ({issue}`15554`)
+* Improve performance of queries with filters or projections on low-cardinality
+  string columns stored in Parquet files. ({issue}`15269`)
+* Fix bug which could lead to more data than necessary getting read from Parquet
```
Comment (martint) [inline docs/src/main/sphinx/release/release-406.md:31]: We generally don't say "Fix bug which ...". Rephrase in terms of what's fixed (e.g., "Fix reading more data than necessary ...")

File: docs/src/main/sphinx/release/release-406.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+  files for queries with filters. ({issue}`15552`)
+* Fix failure when converting JSON statistics to Parquet. ({issue}`15496`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-406.md:33]: This seems like an implementation detail. What are the user visible conditions that cause this failure?

Comment (ebyhr) [reply docs/src/main/sphinx/release/release-406.md:33]: The condition is: 
1. The table has a row within INTEGER range on BIGINT column
2. Checkpoint generation happened (= the operation reached `checkpoint_interval`)

File: docs/src/main/sphinx/release/release-406.md [hunk 14]
```diff
@@ -1,11 +1,13 @@
-# Release 406 (13 Jan 2023)
+# Release 406 (17 Jan 2023)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:1]: 18

File: docs/src/main/sphinx/release/release-406.md [hunk 15]
```diff
@@ -1,11 +1,13 @@
...
 
 ## General
 
 * Add support for [exchange spooling on HDFS](fte-exchange-hdfs) when
   fault-tolerant execution is enabled. ({issue}`15160`)
 * Improve planner estimates for queries containing outer joins over a subquery
   involving `ORDER BY` and `LIMIT`. ({issue}`15428`)
+* Improve accuracy of table scan memory usage reporting. ({issue}`15711`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:9]: of memory usage reporting for table scans

File: docs/src/main/sphinx/release/release-406.md [hunk 16]
```diff
@@ -33,9 +35,11 @@
   filters. ({issue}`15552`)
 * Fix potential query failure when writing to Parquet from a table with an
   `INTEGER` range on a `BIGINT` column. ({issue}`15496`)
+* Fix query failure due to missing null counts in Parquet column indexes. ({issue}`15706`)
 
 ## Hive connector
 
+* Add support for table redirections to the Hudi connector. ({issue}`14750`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:42]: Check how we named this in the past and in the docs .. maybe

to catalogs using the Hudi connector.

File: docs/src/main/sphinx/release/release-406.md [hunk 17]
```diff
@@ -57,16 +62,29 @@
   string columns stored in Parquet files. ({issue}`15269`)
 * Fix bug which could lead to more data than necessary getting read from Parquet
   files for queries with filters. ({issue}`15552`)
+* Fix query failure due to missing null counts in Parquet column indexes. ({issue}`15706`)
 
 ## Iceberg connector
 
+* Add the ID of the Trino query which led to the creation of a new Iceberg
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:69]: Kinda awkward .. but cant think of better/shorter wording atm

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-406.md (hunk 17 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-406.md:69]: Yeah, I spent like 10 minutes on this, and couldn't get it shorter without losing precision.

File: docs/src/main/sphinx/release/release-406.md (hunk 16 above)
Comment (ebyhr) [reply docs/src/main/sphinx/release/release-406.md:42]: The sentence is the same as the past release notes. 
> Add support for table redirections to the Hive connector. ([#11356](https://github.com/trinodb/trino/issues/11356))
https://trino.io/docs/current/release/release-376.html#iceberg-connector

File: docs/src/main/sphinx/release/release-406.md (hunk 17 above)
Comment (ebyhr) [reply docs/src/main/sphinx/release/release-406.md:69]: This is an internal change and there's no need to mention in the release note in my opinion.

File: docs/src/main/sphinx/release/release-406.md [hunk 18]
```diff
@@ -66,6 +70,7 @@
 
 ## Iceberg connector
 
+* Add support for changing column types. ({issue}`15515`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:73]: Is there such a thing as a column type in Iceberg or in general .. or is it the data type of a column .. or is that just a longer name for the same thing

Comment (ebyhr) [reply docs/src/main/sphinx/release/release-406.md:73]: Sorry, I don't understand the question. The entry means supporting `ALTER TABLE [ IF EXISTS ] name ALTER COLUMN column_name SET DATA TYPE new_type` statement in Iceberg connector.

**Comment by elecnix** [comment]
PR description contains this incorrect mention:

@elecnix https://github.com/trinodb/trino/pull/15754 ✅ rn ✅ docs

That PR was by @wendigo

File: docs/src/main/sphinx/release/release-406.md [hunk 19]
```diff
@@ -0,0 +1,96 @@
+# Release 406 (18 Jan 2023)
+
+## General
+
+* Add support for [exchange spooling on HDFS](fte-exchange-hdfs) when
+  fault-tolerant execution is enabled. ({issue}`15160`)
+* Improve planner estimates for queries containing outer joins over a subquery
+  involving `ORDER BY` and `LIMIT`. ({issue}`15428`)
+* Improve accuracy memory usage reporting for table scans. ({issue}`15711`)
```
Comment (hashhar) [inline docs/src/main/sphinx/release/release-406.md:9]:
```suggestion
* Improve accuracy of memory usage reporting for table scans. ({issue}`15711`)
```

Also in general maybe we should add a note regarding all memory usage reporting improvements that it's normal to see "more memory" being used in UI (but same usage from JVM heap) when upgrading Trino?

File: docs/src/main/sphinx/release/release-406.md [hunk 20]
```diff
@@ -0,0 +1,96 @@
...
+* Improve performance of queries parsing date values in ISO 8861 format. ({issue}`15548`)
+* Remove `legacy-phased` execution scheduler. ({issue}`15657`)
```
Comment (hashhar) [inline docs/src/main/sphinx/release/release-406.md:11]: Do mention what configuration needs to be updated if someone was using it for some reason.

File: docs/src/main/sphinx/release/release-406.md [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+* Add support for table redirections to catalogs using the Hudi connector. ({issue}`14750`)
+* Improve performance of reading Parquet files. ({issue}`15498`)
+* Improve memory accounting of the Parquet reader. ({issue}`15554`)
+* Improve performance of queries with filters or projections on low-cardinality
+  string columns stored in Parquet files. ({issue}`15269`)
+* Improve performance of queries with filters when Bloom filter indexes are
+  present in Parquet files. Use of Bloom filters from Parquet files can be
+  disabled with the `parquet.use-bloom-filter` configuration property or the
+  `parquet_use_bloom_filter` session property. ({issue}`14428`)
+* Allow coercion between `UNIONTYPE` and `STRUCT`-typed columns. ({issue}`15017`)
```
Comment (hashhar) [inline docs/src/main/sphinx/release/release-406.md:51]:
```suggestion
* Allow coercion between Hive `UNIONTYPE` and Hive `STRUCT`-typed columns. ({issue}`15017`)
```

i.e. clarify that these are Hive types, not Trino types. To Trino both appear as ROW.

File: docs/src/main/sphinx/release/release-406.md [hunk 22]
```diff
@@ -0,0 +1,96 @@
...
+* Disallow `UPDATE` and `DELETE` on Hive ACID transactional tables when the
+  `deprecated.legacy-update-delete-implementation` configuration property is set
+  to true to prevent correctness issues. ({issue}`12731`)
```
Comment (hashhar) [inline docs/src/main/sphinx/release/release-406.md:54]: This can be removed since it was nullified by https://github.com/trinodb/trino/pull/15161

**Comment by colebow** [comment]
> PR description contains this incorrect mention:
> 
> @elecnix #15754 ✅ rn ✅ docs
> 
> That PR was by @wendigo

Yeah, we mention the maintainer who merged the PR. It was merged by @electrum - I usually type in the first few characters of the maintainer's name and auto-complete with tab. GitHub generally does a good job of prioritizing people who actively are involved in the project, but doing `@elec` and then using tab auto-complete pulls up your name instead of electrum's. I've been manually adjusting for that for months, and I finally messed up in this one. 😞 

I'll be more careful to avoid it happening again.

File: docs/src/main/sphinx/release/release-406.md (hunk 18 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-406.md:76]: Sounds like is just the data type of a column .. all good. We ca leave as it.

File: docs/src/main/sphinx/release/release-406.md (hunk 17 above)
Comment (ebyhr) [reply docs/src/main/sphinx/release/release-406.md:78]: @colebow Not sure why you left this entry. What's the user impact of this?

Comment (colebow) [reply docs/src/main/sphinx/release/release-406.md:78]: @ebyhr didn't see your comment as a reply. Will remove.

File: docs/src/main/sphinx/release/release-406.md [hunk 23]
```diff
@@ -81,6 +85,8 @@
 
 * Add support for changing column types. ({issue}`15515`)
 * Add [support for the JDBC catalog](iceberg-jdbc-catalog). ({issue}`9968`)
+* Add support for changing fields in the `row` type. ({issue}`15808`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-406.md:88]: > Add support for changing fields in columns of `row` type.

BTW, the original PR has this comment for the RN: "with the following suggested text if this PR isn't merged in 406". Does that mean that if we merge it in 406 we don't need RN? Why is that @ebyhr ?

File: docs/src/main/sphinx/release/release-406.md [hunk 24]
```diff
@@ -0,0 +1,118 @@
+# Release 406 (24 Jan 2023)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:1]: 25

File: docs/src/main/sphinx/release/release-406.md [hunk 25]
```diff
@@ -0,0 +1,118 @@
...
+  table has recently been written to. ({issue}`14981`)
+
+## Cassandra connector
+
+* Fix incorrect results when the Cassandra `list`, `map`, or `set` types contain
+  user-defined types. ({issue}`15771`)
+
+## Delta Lake connector
+
+* Improve latency for `INSERT` queries on unpartitioned tables. ({issue}`15708`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:43]: Reduce latency ?

File: docs/src/main/sphinx/release/release-406.md [hunk 26]
```diff
@@ -0,0 +1,118 @@
...
+* Fix reading more data than necessary from Parquet files for queries with
+  filters. ({issue}`15552`)
+* Fix potential query failure when writing to Parquet from a table with an
+  `INTEGER` range on a `BIGINT` column. ({issue}`15496`)
+* Fix query failure due to missing null counts in Parquet column indexes. ({issue}`15706`)
+
+## Hive connector
+
+* Add support for table redirections to catalogs using the Hudi connector. ({issue}`14750`)
+* Improve latency for `INSERT` queries on unpartitioned tables. ({issue}`15708`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:57]: Reduce ?

File: docs/src/main/sphinx/release/release-406.md [hunk 27]
```diff
@@ -0,0 +1,118 @@
...
+* Fix reading more data than necessary from Parquet files for queries with
+  filters. ({issue}`15552`)
+* Fix query failure due to missing null counts in Parquet column indexes. ({issue}`15706`)
+
+## Iceberg connector
+
+* Add support for changing column types. ({issue}`15515`)
+* Add [support for the JDBC catalog](iceberg-jdbc-catalog). ({issue}`9968`)
+* Add support for changing fields in the `row` type. ({issue}`15808`)
+* Improve latency for `INSERT` queries on unpartitioned tables. ({issue}`15708`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-406.md:89]: Reduce

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-406.md (hunk 23 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-406.md:88]: Make no sense to me. I would just leave it in and ship the release. If really necessary we can take out after .. although I dont understand why that would be the case..

File: docs/src/main/sphinx/release/release-406.md [hunk 28]
```diff
@@ -0,0 +1,117 @@
...
+  string columns stored in Parquet files. ({issue}`15269`)
+* Fix reading more data than necessary from Parquet files for queries with
+  filters. ({issue}`15552`)
+* Fix query failure due to missing null counts in Parquet column indexes. ({issue}`15706`)
+
+## Iceberg connector
+
+* Add support for changing column types. ({issue}`15515`)
+* Add [support for the JDBC catalog](iceberg-jdbc-catalog). ({issue}`9968`)
+* ReduceAdd support for changing fields in columns of row type. latency for `INSERT` queries on unpartitioned tables. ({issue}`15708`)
```
Comment (ebyhr) [inline docs/src/main/sphinx/release/release-406.md:88]: This line is broken.

File: docs/src/main/sphinx/release/release-406.md (hunk 23 above)
Comment (ebyhr) [reply docs/src/main/sphinx/release/release-406.md]: This entry is unnecessary becasue it's already covered by another entry:
```
Add support for changing column types. ({issue}`15515`)
```

File: docs/src/main/sphinx/release/release-406.md [hunk 29]
```diff
@@ -0,0 +1,117 @@
...
+
+## General
+
+* Add support for [exchange spooling on HDFS](fte-exchange-hdfs) when
+  fault-tolerant execution is enabled. ({issue}`15160`)
+* Add support for `CHECK` constraints in an `INSERT` statement. ({issue}`14964`)
+* Improve planner estimates for queries containing outer joins over a subquery
+  involving `ORDER BY` and `LIMIT`. ({issue}`15428`)
+* Improve accuracy of memory usage reporting for table scans. ({issue}`15711`)
+* Improve performance of queries parsing date values in ISO 8861 format. ({issue}`15548`)
```
Comment (ebyhr) [inline docs/src/main/sphinx/release/release-406.md:11]: Change to ISO 8601.
